packages/shared/examples/range-plot/range-plot.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useCallback } from 'react'
+import { Scale, Scatter } from '@unovis/ts'
+import { VisXYContainer, VisBulletLegend, VisTooltip, VisAxis, VisScatter, VisLine } from '@unovis/react'
+
+import { data, DataRecord, processLineData } from './data'
+
+const height = 1600
+const yScale = Scale.scalePoint([0, 800]).domain(data.map(d => d.occupation))
+const lineData = processLineData(data)
+
+export default function RangePlot (): JSX.Element {
+  const legendItems = [{ name: 'Women', color: '#FF6B7E' }, { name: 'Men', color: '#4D8CFD' }]
```

Comment:
If we want to show how to use Unovis components efficiently with React, you should use `useMemo` or `useCallback` everywhere we initialize a new object, array, or function.